### PR TITLE
feat: time weighted average

### PR DESCRIPTION
**Pull request by jlapacik**
### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

File: stdlib/universe/universe.flux [hunk 1]
```diff
@@ -98,6 +98,11 @@ builtin logarithmicBins
 builtin sleep // sleep is the identity function with the side effect of delaying execution by a specified duration
 builtin die // die returns a fatal error from within a flux script
 
+// Time weighted average where values at the beginning and end of the range are linearly interpolated.
+timeWeightedAvg = (tables=<-, unit) => tables
+    |> integral(unit: unit, interpolate: "linear")
+    |> map(fn: (r) => ({ r with _value: (r._value * float(v: uint(v: unit))) / float(v: int(v: r._stop) - int(v: r._start)) }))
```
Comment (rockstar) [inline stdlib/universe/universe.flux:104]: When I opened this patch, I expected there to be a chunk of go code. _Very_ happy to see it implemented in pure flux.

File: stdlib/universe/universe.flux [hunk 2]
```diff
@@ -98,6 +98,11 @@ builtin logarithmicBins
 builtin sleep // sleep is the identity function with the side effect of delaying execution by a specified duration
 builtin die // die returns a fatal error from within a flux script
 
+// Time weighted average where values at the beginning and end of the range are linearly interpolated.
```
Comment (rockstar) [inline stdlib/universe/universe.flux:101]: Was there an explicit decision to put this in universe? I'm starting to be worried that we might be polluting the namespace a lot now.

File: stdlib/universe/universe.flux (hunk 1 above)
Comment (jlapacik) [reply stdlib/universe/universe.flux:104]: Go code was needed for `integral` https://github.com/influxdata/flux/pull/3146.

File: stdlib/universe/universe.flux (hunk 2 above)
Comment (jlapacik) [reply stdlib/universe/universe.flux:101]: No explicit decision, I just wasn't sure what package this function should belong to. I agree about polluting the universe package though. I'm open to any suggestions.

**Pull request merged**